frontend/src/app/admin/company/create-company/create-company.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Observable } from 'rxjs';
+
+import { CommonModule } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, FormsModule, ReactiveFormsModule, Validators } from '@angular/forms';
+import { RouterModule } from '@angular/router';
+
+import { LetDirective } from '@app/common/directives/let/let.directive';
+import { FormErrorMessagePipe } from '@app/common/pipes/form-error-message/form-error-message.pipe';
+
+import { CreateCompanyFormGroup } from './common/models/create-company.model';
+import { CreateCompanyState, CreateCompanyStore } from './create-company.store';
+
+@Component({
+    selector: 'app-create-company',
+    standalone: true,
+    imports: [CommonModule, FormsModule, ReactiveFormsModule, FormErrorMessagePipe, RouterModule, LetDirective],
+    providers: [CreateCompanyStore],
+    templateUrl: './create-company.component.html'
+})
+export class CreateCompanyComponent implements OnInit {
+    form!: FormGroup<CreateCompanyFormGroup>;
+
+    vm$: Observable<CreateCompanyState> = this.createCompanyStore.vm$;
+
+    selectedFile: File | null = null;
+
+    constructor(
+        private readonly fb: FormBuilder,
+        private readonly createCompanyStore: CreateCompanyStore
+    ) {}
+
+    ngOnInit(): void {
+        this.initForm();
+    }
+
+    onSubmit(): void {
+        const formData = new FormData();
+        for (const key of Object.keys(this.form.controls)) {
```

Comment:
instead of going through the controls do something like:
```ts
const formValue = this.form.getRawValue();
for (const key of Object.keys(formValue)) {
   formData.append(key, formValue[key as keyof formValue]);
}